airflow/api_connexion/schemas/task_instance_schema.py

```diff
@@ -122,7 +122,8 @@ class SetTaskInstanceStateFormSchema(Schema):
 
     dry_run = fields.Boolean(dump_default=True)
     task_id = fields.Str(required=True)
-    execution_date = fields.DateTime(required=True, validate=validate_istimezone)
+    execution_date = fields.DateTime(validate=validate_istimezone)
+    dag_run_id = fields.Str()
```

Comment:
Is it possible to add a custom validation so that this is an either/or thing? (I don't think passing both should be allowed)

Reply:
The openAPi spec would have been a better place to handle it but it doesn't support it.

We can do it here but it won't appear in the documentation